packages/syft/src/syft/core/node/new/user_code.py

```diff
@@ -43,11 +55,16 @@ class InputPolicy(SyftObject):
     __canonical_name__ = "InputPolicy"
     __version__ = SYFT_OBJECT_VERSION_1
 
+    # relative
+    from .api import UserNodeView
+
     id: UID
-    inputs: Dict[str, Any]
+    inputs: Dict[UserNodeView, Any]
```

Comment:
The input policy key is the string of the kwarg name when invoking the function. What is the intention of using UserNodeView here instead?

Reply:
Currently we modify the input notation to also include the party it originated from.....

Assume we performing computation in Canada domain , we require access to the variable `data_1` in the data set

In the previous notations

```py

inputs = {"data_1" : UID() }
```

In the new notation:

```py
inputs = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : {"data_1" : UID() } }
```

Sort of like a nested dict, which maps the data from the parties it originated from , 

Consider two domains `Canada` , `Italy`
```py
inputs = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : {"data_1" : UID() },
                 UserNodeView(name="Italy", verify_key= canada_domain.verify_key) : {"data_2" : UID() },
 }
```